WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -492,6 +488,31 @@ private void updateConnectionBar(boolean isConnected) {
         }
     }
 
+    private void handleBlogRemoved() {
+        if (!AccountHelper.isSignedIn()) {
+            ActivityLauncher.showSignInForResult(this);
+        } else {
+            MySiteFragment mySiteFragment = getMySiteFragment();
+            if (mySiteFragment != null) {
+                List<Map<String, Object>> allBlogs = WordPress.wpDB.getAllBlogs();
+                if (allBlogs.isEmpty()) {
+                    mySiteFragment.setBlog(null);
+                } else {
+                    showSitePickerWithCurrentBlog();
+                }
+            }
+        }
+    }
+
+    private void showSitePickerWithCurrentBlog() {
```

Comment:
Would it make sense to move this helper method to `ActivityLauncher` in case we need to do the same thing somewhere else?